src/script/event/EventRepository.ts

```diff
@@ -524,15 +524,28 @@ export class EventRepository {
     const newEventData = newEvent.data;
     // the preview status is not sent by the client so we fake a 'preview' status in order to cleanly handle it in the switch statement
     const ASSET_PREVIEW = 'preview';
+    // similarly, no status is sent by the client when we retry sending a failed message
+    const RETRY_EVENT = 'retry';
     const isPreviewEvent = !newEventData.status && !!newEventData.preview_key;
-    const previewStatus = isPreviewEvent ? ASSET_PREVIEW : newEventData.status;
+    const isRetryEvent = !!newEventData.content_length;
+    const handledEvent = isRetryEvent ? RETRY_EVENT : newEventData.status;
+    const previewStatus = isPreviewEvent ? ASSET_PREVIEW : handledEvent;
+
+    const updateEvent = () => {
+      const updatedData = {...originalEvent.data, ...newEventData};
+      const updatedEvent = {...originalEvent, data: updatedData};
+      return this.eventService.replaceEvent(updatedEvent);
+    };
 
     switch (previewStatus) {
       case ASSET_PREVIEW:
+
+      case RETRY_EVENT: {
+        return updateEvent();
+      }
+
```

Comment:
you could also do like this ;) 

```suggestion
      case RETRY_EVENT: 
```

up to you what you prefer